xmodule/docs/decisions/0003-library-content-block-schema.rst

```diff
@@ -0,0 +1,201 @@
+
+   * - source_library\_ version
+     - settings, str|None
+     - None
+     - TBD
+     - Version of V1 source library (a MongoID string). None means latest.
+
+   * - mode
+     - settings, str
+     - "random"
+     - "random"
+     - How blocks should be selected. Only valid value is "random".
+
+   * - capa_type
+     - settings, str
+     - "any"
+     - "symbolicresponse"
+     - Unless "any": Include only problems, and only problems of this response type.
+
+   * - children
+     - special, list of XBlocks
+     - []
+     - ``block-v1:...+type@problem+block@A``
+       ``block-v1:...+type@problem+block@B``
+       ``block-v1:...+type@problem+block@D``
+       ``block-v1:...+type@problem+block@F``
+     - Copy of library contents, filtered by capa_type. Course author may apply edits that are local to this library_content block usage. Edits to child blocks' settings persist across library version upgrades. Edits to child blocks' content are wiped out when library version is upgraded.
+
+   * - max_count
+     - settings, int
+     - 1
+     - 2
+     - Number of children to randomly select for each learner. -1 means all.
+
+   * - selected
+     - user_state, list of (block type, block id) tuples
+     - []
+     - [("problem", "A"), ("problem", "F")]
+     - An ordered sub-list of the children, to be shown to a given learner. Once populated, this is kept stable for the learner, unless a the children or max_count are changed, in which case it will have blocks added or removed as necessary. If the list changes, it is re-shuffled.
+
+
+Decision
+********
+
+We will introduce the support for a "manual selection mode", without explicitly making it *mode* as hinted at by the old block schema. This will lead to a cleaner and more flexible implementation. We will achieve this with two new boolean settings: **manual** and **shuffle**:
+
+* When manual is *disabled* and shuffle is *enabled*, the block will behave as it did before. That is, it will import the entire library (filtered by capa_type) as its children, and present each user a random subset based on max_count. For backwards compatibility, these will be the default value of the settings.
+
+* When manual is *enabled*, shuffle is *disabled*, and max_count is set to -1, the block will behave in the "static" mode. That is, the user will be prompted to select specific blocks in the library, and each user will be presented those blocks in order.
+
+* When manual is *enabled*, any filter fields (currently just capa_type, but perhaps more in the future) will be ignored for the purposes of CMS deciding which blocks to import. That's because in the event that manually-picked children clash with the filters, we need to decide who would "win", and we are deciding here that the manually-picked children would win. However, we could choose to have the block-picker filter blocks based on filters.
+
+We will also remove the **mode** field, as it is no longer needed.
```

Comment:
Are we saying it's safe to deprecate this field because it's only ever had one valid value?

Reply:
Yes